ansible/roles/debops.postfix_virtual_ldap/defaults/main.yml

```diff
@@ -0,0 +1,685 @@
+# under which the accesses of Postfix, Dovecot and other components of the mail server should take place.
+# On the one hand this prevents mail server components from accessing sensitive system directories,
+# on the other hand it protects the mailboxes from external access.
+# Only vmail (and root) are allowed to access the mailboxes.
+postfix_virtual_ldap__vmail_posix_user: 'vmail'
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__vmail_posix_uidnumber [[[
+#
+# Virtual Mail POSIX uidNumber
+#
+postfix_virtual_ldap__vmail_posix_uidnumber: '{{ getent_passwd[postfix_virtual_ldap__vmail_posix_user][1]|int }}'
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__vmail_posix_group [[[
+#
+# Virtual Mail POSIX group
+#
+postfix_virtual_ldap__vmail_posix_group: 'vmail'
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__vmail_posix_gidnumber [[[
+#
+# Virtual Mail POSIX gidNumber
+#
+postfix_virtual_ldap__vmail_posix_gidnumber: '{{ getent_group[postfix_virtual_ldap__vmail_posix_group][1]|int }}'
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__mailbox_base [[[
+#
+# All mailboxes are stored directly in the file system of the server.
+#
+# This setting describes the prefix that the virtual delivery agent prepends to all pathname results from
+# `$virtual_mailbox_maps` table lookups.
+# The directories under this base contain the users MailDirs.
+# This directory must be created in advance and must be writeable by all potential mail users.
+# To achieve this, either use the same GID across all mail users and set group ownership to that GID,
+# or make the directory world writable.
+postfix_virtual_ldap__mailbox_base: '/var/vmail'
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__system_users [[[
+#
+# Postfix Virtual Users
+postfix_virtual_ldap__system_users:
+  - name: '{{ postfix_virtual_ldap__vmail_posix_user }}'
+    group: '{{ postfix_virtual_ldap__vmail_posix_group }}'
+    home: '{{ postfix_virtual_ldap__mailbox_base }}'
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__virtual_alias_maps [[[
+#
+# Optional lookup tables that alias specific mail addresses or domains
+# to other local or remote address.
+# The default option, when using Virtual Mail with LDAP, will first go over the
+# alisaes set by :ref:`debops.etc_aliases` and if no match is found query
+# the LDAP for the email address.
+#
+# Specify zero or more `type:name` lookup tables, separated by whitespace or comma.
+# Tables will be searched in the specified order until a match is found.
+# Note: these lookups are recursive.
+postfix_virtual_ldap__virtual_alias_maps:
+  - '$alias_maps'
```

Comment:
Actually it solved things. 
Without this line emails to any email/alias in `/etc/aliasies` will be dropped or sent to a catch-all address.
Therefore the `$alias_maps` is required here on the first position, in order for postfix to find a match first in an alias and when not found in the LDAP mails or aliases.

Reply:
Does anyone else, familiar with postfix, support this config?

Reply:
Perhaps [this thread on ServerFault](https://serverfault.com/questions/644306/) can shine some light on this. Perhaps you could show an example from `/etc/aliases` that resulted in mail being dropped when `$alias_maps` was not included in `$virtual_alias_maps`? I could then test this in my development environment, would be very interesting to know this behaviour.

Reply:
As of [Confused about alias_maps and virtual_alias_maps](https://serverfault.com/a/644393):

> Parameter virtual_alias_maps
>> Always invoked first time before any other address classes. It doesn't care whether the recipient domain was listed in mydestination, virtual_mailbox_domains or other places. It will override the address/alias defined in other places.

So this means, that if  we have users in LDAP with the following attributes:
- Alice:
  - `mail` alice.dow@example.net
  - `mailAlias: admin@example.net`

- Bob:
  - `mail` bob.jones@example.net
  - `mailAlias: @example.net`
  - `mailAlias: bob@anotherdomain.com`
  - `mailAlias: webmaster@anotherdomain.com`

 And this `/etc/aliases`:
```
root:                                admin@example.net
webmaster:                     admin@example.net
foo:                                  carl@othermailprovider.com
```

Using my proposed configuration, the following recipients will be routed as follows:
- webmaster --> Alias:admin@example.net --> Mailbox:alice.dow@example.net
- webmaster@example.com --> Alias:admin@example.net --> Mailbox:alice.dow@example.net
- alice@example.net -->CatchAll-Alias:@example.net --> Mailbox:bob.jones@example.net
- a@example.net -->CatchAll-Alias:@example.net --> Mailbox:bob.jones@example.net
- webmaster@anotherdomain.com -> Mailbox:bob.jones@example.net
- foo --> MailBox: carl@othermailprovider.com

Without explicitly listing  `$alias_maps` in `virtual_alias_maps` , `virtual_alias_maps` will just ask the LDAP server for `mail` and `mailAlias` without querying the local aliases database.